src/components/Table.js

```diff
@@ -0,0 +1,202 @@
+import classnames from 'classnames';
+import { useEffect, useRef } from 'preact/hooks';
+
+import { refElement } from '../util/typing';
+
+import { Scrollbox } from './containers';
+import { Spinner } from './Spinner';
+
+/**
+ * @typedef TableColumn
+ * @prop {string} label - Header label for the column
+ * @prop {string} [classes] - Additional CSS classes for the column's `<th>` element
+ */
+
+/**
+ * @template Item
+ * @typedef TableProps
+ * @prop {string} accessibleLabel - An accessible label for the table
+ * @prop {string} [classes] - Extra CSS classes to apply to the outermost
+ *   element, which is a <Scrollbox> div
+ * @prop {TableColumn[]} columns - The columns to display in this table
+ * @prop {boolean} [isLoading] - Show an indicator that data for the table is
+ *   currently being fetched
+ * @prop {Item[]} items -
+ *   The items to display in this table, one per row. `renderItem` defines how
+ *   information from each item is represented as a series of table cells.
+ * @prop {(it: Item, selected: boolean) => any} renderItem -
+ *   A function called to render each item as the contents of a table row.
+ *   The result should be a list of `<td>` elements (one per column) wrapped inside a Fragment.
+ * @prop {Item|null} selectedItem - The currently selected item from `items`
+ * @prop {(it: Item) => void} onSelectItem -
```

Comment:
I agree that there could be future use cases in which `onSelectItem` and/or `onUseItem` may not be relevant, but as we discussed in the call, I'd like to kick this can down the road a bit until we have a use case. I'd like to leave `onSelectItem` as-is for the moment; I think it's a generally OK, if not perfect name, and changing it would be an API breaking change, and would have a bit of a cascade effect (e.g. there is an `.is-selected` class). At present, I think "selected" and "focused" are roughly interchangeable in the `Table` for a given `item`. If this changes—if logic evolves such that there is a meaningful logical difference between a focused item and a "selected" one, then I think we can address the naming again...